src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Wasm/WasmBase.cs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+
+namespace System.Runtime.Intrinsics.Wasm
+{
+    [Intrinsic]
+    internal abstract class WasmBase
```

Comment:
Why not just handle `BitOperations.LeadingZeroCount` and `BitOperations.TrailingZeroCount` directly?

Reply:
I wanted to stay close to the other platforms implementation.

Reply:
The others exist largely due to historical reasons and because on x86 they aren't part of the "baseline" instruction set. There will ultimately be better codegen and less work for the compiler if you recognize the `BitOperations` APIs directly. This is because the compiler won't have to do inlining, dead code elimination for `n` platforms, etc.

I'm planning on doing the same for RyuJIT and recognizing `BitOperations` directly for the same reasons.

Reply:
What do you plan to do with the Log2 methods? Implement them as direct intrinsics too?

Reply:
For `Log2` yes. For `RoundUpToPowerOf2` that's TBD, but potentically.

tzcnt and lzcnt are the most hot methods and the most impactful in terms of "compiler work" required to inline and optimize down to a single instruction today.